src/Elmish/React/CommonProps.purs

```diff
@@ -0,0 +1,41 @@
+module Elmish.React.CommonProps
+    ( CommonProps
+    , withCommonProps
+    , withKey
+    , withPropsUnsafe
+    ) where
+
+import Prelude
+
+import Elmish.React (class ValidReactProps, ReactElement, cloneElement)
+import Elmish.React.Import (class IsSubsetOf)
+
+-- | Row of props that are common to all React components, without having to
+-- | declare them.
+type CommonProps = ( key :: String )
```

Comment:
**Q:** Since `key` is truly universal in React, why not keep it as default `CommonProps`? I wonder because `key` is a performance optimization, so when you actually want to use it, the overhead `React.cloneElement` seems a bit ironic.

---

BTW, I was curious how come we never ran into `key` warnings and the reason is because most of our ‘lists’ are passed in as `children` for which we have a special case:
https://github.com/collegevine/purescript-elmish/blob/4ddcb8e12ec695e022a5c7212add49e8c377a5cc/src/Elmish/React.js#L26-L40

When I tried with passing a non-`children` prop, I got the warning, e.g.

**Test.purs**
```purescript
module JSX.Test
    ( testKey
    , TestKeyProps
    ) where

import Elmish.React (ReactElement, createElement')
import Elmish.React.Import (EmptyProps, ImportedReactComponent, ImportedReactComponentConstructor)

type TestKeyProps r =
    ( items :: Array ReactElement
    | r
    )

testKey :: ImportedReactComponentConstructor TestKeyProps EmptyProps
testKey = createElement' testKey_
foreign import testKey_ :: ImportedReactComponent
```

**Test.js**
```javascript
exports.testKey_ = require("./Test.jsx").default
```

**Test.jsx**
```
import React from "react"

export default ({ items }) => <ul>{items}</ul>
```

**Main.purs**
```
testKey
    { items:
        [ H.h1 {} "foo"
        , H.h1 {} "bar"
        , H.h1 {} "baz"
        ]
    }
```

![image](https://user-images.githubusercontent.com/49583/57085057-747baf80-6cc9-11e9-8fb6-31c817df86f8.png)